DigitalLearningSolutions.Data/Services/DelegateUploadFileService.cs

```diff
@@ -203,6 +211,15 @@ private void RegisterDelegate(DelegateTableRow delegateRow, DateTime? welcomeEma
                     );
                 default:
                     var newDelegateRecord = userDataService.GetDelegateUserByCandidateNumber(errorCodeOrCandidateNumber, centreId)!;
+                    if (welcomeEmailDate.HasValue)
+                    {
+                        passwordResetService.GenerateAndScheduleDelegateWelcomeEmail(

```

Comment:
I'm worried that this method might not have originally been set up correctly. 

It performs all its logic based on a delegate user that it gets from:
```
            var delegateUsers = userService.GetDelegateUsersByEmailAddress(recipientEmailAddress);
            var delegateUser = delegateUsers.FirstOrDefault() 
```
But if there are already delegate accounts with the email address at other centres I think the delegateUser will not be the newly created one. The email generated will probably state an incorrect candidate number,  and the password hash will be set to the wrong account in the database. I'm not sure if the setting of passwords is account specific or if it will reset the password for all of them, so that might not be a concern. 
Do you know if this works as expected when there are multiple accounts with the same email?

Reply:
I've raised https://softwiretech.atlassian.net/browse/HEEDLS-642 for this, but done the code fix in this MR. (I will move across the ticket when it reaches test.)